common/report.c

```diff
@@ -54,9 +54,12 @@ static oe_result_t _oe_parse_sgx_report_body(
             report_body->mrsigner,
             sizeof(report_body->mrsigner)));
 
-    parsed_report->identity.product_id[0] = report_body->isvprodid & 0xFF;
+    if (report_body->isvprodid > OE_INT8_MAX)
+        goto done;
+    parsed_report->identity.product_id[0] =
+        (uint8_t)report_body->isvprodid & 0xFF;
     parsed_report->identity.product_id[1] =
-        (report_body->isvprodid >> 8) & 0xFF;
+        ((uint8_t)report_body->isvprodid >> 8) & 0xFF;
```

Comment:
Same bug here as #1064 and #1065

Reply:
@berinpaul Oh okay, sorry I didn't read all the comments in this conversation. Should we close this PR then?